Src/BinaryConverter/Serializers/DateTimeSerializer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BinaryConverter.Serializers
+{
+    public class DateTimeSerializerArg : ISerializerArg
+    {
+        public long TickResolution { get; set; } = 0;
+    }
+
+    public class DateTimeSerializer : BaseSerializer
+    {
+        public override object Deserialize(BinaryTypesReader br, Type type, SerializerSettings settings, ISerializerArg serializerArg)
+        {
+            var typedSerializerArg = GetSerializerArg<DateTimeSerializerArg>(type, settings, serializerArg);
+
+            if (typedSerializerArg != null && typedSerializerArg.TickResolution > 0)
+            {
```

Comment:
Maybe it is possible to not branch every Serialize/Deserialize method into two different cases: 

You can use null propagation as in example below, or you can introduce a default serializer arg, so that the arg is always passed, but sometimes its a custom one, and sometimes its a default.

```c#
public override object Deserialize(BinaryTypesReader br, Type type, SerializerSettings settings, ISerializerArg serializerArg)
{
    var typedSerializerArg = GetSerializerArg<DateTimeSerializerArg>(type, settings, serializerArg);
    return br.ReadCompactDateTime(typedSerializerArg?.TickResolution);
    // BTW, why DateTimeKind.Utc was used here, only when serializer arg wasn't passed?
}
```
and
```c#
public DateTime ReadCompactDateTime(long? ticksResolution)
{
    // Why DateTimeKind.Utc is not passed here?
    return new DateTime(Read7BitLong() * (ticksResolution ?? 1));
}
```

(This comment applies to all serializers, not just DataTime).